pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -170,6 +170,21 @@ type Matrix struct {
 	// The names of the `params` in the `Matrix` must match the names of the `params` in the underlying `Task` that they will be substituting.
 	// +listType=atomic
 	Params []Param `json:"params,omitempty"`
+
+	// Include is a list of MatrixInclude
+	// Note this field is in preview mode and not yet supported
```

Comment:
thank you @EmmaMunley for this PR, it looks good to me!

The build is failing with:

```
ERROR: New API rule violations found which are not present in hack/ignored-openapi-violations.list. Please fix these violations:
API rule violation: list_type_missing,github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1,Matrix,Include
API rule violation: list_type_missing,github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1,MatrixInclude,Params
```

I think we are missing an annotation here:

```
// +listType=atomic
```

Reply:
Thanks for the catch!